recipes-omnect/initrdscripts/omnect-os-initramfs/grub-sh

```diff
@@ -71,3 +71,70 @@ get_fsck_status()
     fi
     echo -n "${return}"
 }
+
+
+rootblk_dev_generate_dev_omnect() {
+    local search_rootblk=("/dev/sda" "/dev/nvme0n1p")
+    local rootblk=""
+    local rootpart=${bootparam_rootpart}
+    local part=""
+
+    [ -z "${bootparam_rootpart}" ] && msg "no rootpart set" && return 1
+
+    mkdir -p /dev/omnect
+    cd /dev/omnect
+
+    # get rootblk via kernel param ... if not set, we search
+    rootblk=${bootparam_rootblk}
+    msg "env rootblk=${bootparam_rootblk}"
+    if [ -z "${rootblk}" ]; then
+        for i in "${search_rootblk[@]}"
+        do
+            msg "probing \"${i}${rootpart}\""
+            # we wait for max 2sec - make it configurable?
+            local sleep_cnt=0
+            while [ ! -b "${i}${rootpart}" ] && [ ${sleep_cnt} -le 20 ]; do
+                debug "waiting for \"${i}${rootpart}\""
+                sleep 0.1
+                sleep_cnt=$((${sleep_cnt}+1))
+            done
+
+            if [ -b "${i}${rootpart}" ]; then
+                rootblk="${i}"
+                msg "found \"${rootblk}\""
+                break;
+            fi
+        done
+    else
+        while [ ! -b "${rootblk}1" ]; do
+            debug "waiting for rootblk=${rootblk}"
+            om 0.1
```

Comment:
what is om?
perhaps should that be similar to "while" loop above? (lines 96 - 100)

Reply:
done